projects/BFDR/NatalityRecord_constructors.cs

```diff
@@ -221,40 +233,71 @@ public Bundle GetDemographicCodedContentBundle()
+            // Note: Unlike mortality records, where some bundles are collections so don't have a composition,
+            // all natality records are documents and so should have a composition
+            Composition = (Composition) Bundle.Entry.FirstOrDefault(entry => entry.Resource is Composition)?.Resource;
+            if (Composition == null)
             {
                 throw new System.ArgumentException("Failed to find a Composition. The first entry in the FHIR Bundle should be a Composition.");
             }
+
+            // See if this is a type of bundle that requires a subject
+            bool requiresSubject = COMPOSITIONS_REQUIRING_SUBJECT.Contains(Composition.Type?.Coding?[0]?.Code);
+
             // Depending on the type of bundle, some of this information may not be present, so check it in a null-safe way
-            if (fullRecord && (Composition.Subject == null || String.IsNullOrWhiteSpace(Composition.Subject.Reference)))
+            if (requiresSubject && String.IsNullOrWhiteSpace(Composition.Subject?.Reference))
             {
-                throw new System.ArgumentException("The Composition is missing a subject (a reference to the Child resource).");
+                throw new System.ArgumentException("The Composition is missing a subject reference to the Child resource.");
             }
-            // Grab Subject (Child/Fetus) and Mother.
+
+            // Retrieve Subject (Child/Fetus)
             List<Patient> patients = Bundle.Entry.FindAll(entry => entry.Resource is Patient).ConvertAll(entry => (Patient) entry.Resource);
-            string subjectId = Composition?.Subject.Reference;
-            Subject = patients.Find(patient => patient.Meta.Profile.Any(p => p == subjectProfile) && subjectId.Contains(patient.Id));
-            Mother = patients.Find(patient => patient.Meta.Profile.Any(p => p == VR.ProfileURL.Mother));
-            // Grab Father
-            Father = Bundle.Entry.FindAll(entry => entry.Resource is RelatedPerson).ConvertAll(entry => (RelatedPerson) entry.Resource).Find(resource => resource.Meta.Profile.Any(relatedPersonProfile => relatedPersonProfile == VR.ProfileURL.RelatedPersonFatherNatural));
-            Coverage = Bundle.Entry.FindAll(entry => entry.Resource is Coverage).ConvertAll(entry => (Coverage) entry.Resource).Find(resource => resource.Meta.Profile.Any(coverageProfile => coverageProfile == ProfileURL.CoveragePrincipalPayerDelivery));
-            // Grab attendant and certifier
-            List<Practitioner> practitioners = Bundle.Entry.FindAll(entry => entry.Resource is Practitioner).ConvertAll(entry => (Practitioner) entry.Resource);
-            Attendant = practitioners.Find(patient => patient.Extension.Any(ext => Convert.ToString(ext.Value) == "attendant"));
-            Certifier = practitioners.Find(patient => patient.Extension.Any(ext => Convert.ToString(ext.Value) == "certifier"));
+            string subjectId = Composition.Subject?.Reference;
```

Comment:
This approach looks good, just want to make sure I understand. The composition subject is the child or fetus so you're using the subject id to identify which patient is the child vs. the patient mother?

Reply:
Yes, this identifier lets us find the child/fetus patient entry and we use an identifier from a different reference to identify the mother patient entry. My first attempt assumed that the mother was whichever patient wasn't the child, which could be an alternative but seemed less precise.